extensions/blocks/recurring-payments/edit.jsx

```diff
@@ -412,116 +523,31 @@ class MembershipsButtonEdit extends Component {
-							<Spinner />
-						</Placeholder>
-					) }
-				{ ! this.state.shouldUpgrade &&
-					! this.props.attributes.planId &&
-					connected === API_STATE_CONNECTED &&
-					products.length === 0 && (
-						<div className="wp-block-jetpack-recurring-payments">
-							<Placeholder
-								icon={ <BlockIcon icon={ icon } /> }
-								label={ __( 'Payments', 'jetpack' ) }
-								notices={ notices }
-							>
-								<div className="components-placeholder__instructions">
-									<p>
-										{ __( 'To use this block, first add at least one payment plan.', 'jetpack' ) }
-									</p>
-									{ this.renderAddMembershipAmount( true ) }
-								</div>
-							</Placeholder>
-						</div>
-					) }
-				{ ! this.state.shouldUpgrade &&
-					! this.props.attributes.planId &&
-					this.state.addingMembershipAmount !== PRODUCT_FORM_SUBMITTED &&
-					connected === API_STATE_CONNECTED &&
-					products.length > 0 && (
-						<div className="wp-block-jetpack-recurring-payments">
-							<Placeholder
-								icon={ <BlockIcon icon={ icon } /> }
-								label={ __( 'Payments', 'jetpack' ) }
-								notices={ notices }
-							>
-								<div className="components-placeholder__instructions">
-									<p>
-										{ __(
-											'To use this block, select a previously created payment plan.',
-											'jetpack'
-										) }
-									</p>
-									{ this.renderMembershipAmounts() }
-									<p>{ __( 'Or a new one.', 'jetpack' ) }</p>
-									{ this.renderAddMembershipAmount( false ) }
-								</div>
-							</Placeholder>
-						</div>
-					) }
-				<InspectorControls>
-					<PanelBody title={ __( 'Payments & Fees', 'jetpack' ) }>
-						<ExternalLink href="https://wordpress.com/support/wordpress-editor/blocks/payments/#related-fees">
-							{ __( 'Read more about Payments and related fees.', 'jetpack' ) }
-						</ExternalLink>
-					</PanelBody>
-				</InspectorControls>
+				{ ! this.isPremiumContentChild && this.renderUpgradeNudges() }
+				{ ! this.isPremiumContentChild && this.renderPlanNotices() }
+
 				{ showControls && inspectorControls }
-				{ ( ( ( this.hasUpgradeNudge || ! this.state.shouldUpgrade ) &&
-					connected !== API_STATE_LOADING ) ||
-					this.props.attributes.planId ) && (
-					<InnerBlocks
-						template={ [
-							[
-								'jetpack/button',
-								{
-									element: 'a',
-									uniqueId: 'recurring-payments-id',
-								},
-							],
-						] }
-						templateLock="all"
-					/>
-				) }
+
+				<InnerBlocks
+					template={ [
+						[
+							'jetpack/button',
+							{
+								element: 'a',
+								uniqueId: 'recurring-payments-id',
+							},
+						],
+					] }
+					templateLock="all"
+				/>
 			</Fragment>
 		);
 	};
 }
 
 export default compose( [
-	withSelect( ( select, { clientId } ) => ( {
+	withSelect( ( select ) => ( {
 		postId: select( 'core/editor' ).getCurrentPostId(),
-		innerButtons: select( 'core/editor' ).getBlocksByClientId( clientId ),
```

Comment:
I think this may be causing issues when trying to assign a payment plan to a button? I get a `Uncaught TypeError: Cannot read property 'length' of undefined` error when I try to do so right now. 

![Screen Recording 2020-11-25 at 07 15 45 PM](https://user-images.githubusercontent.com/426388/100266810-b7a80e00-2f52-11eb-9592-0d67aebb0f82.gif)

Reply:
Good catch, this was not a UI status I'd seen, will look into it.